backend/src/main/java/com/example/userservice/controller/RegisteredUserController.java

```diff
@@ -24,7 +22,9 @@ public RegisteredUser addUser(@RequestBody RegisteredUser registeredUser){
     }
 
     @Override
-    public RegisteredUser getUser(Integer id) {
-        return userService.getUser(id);
+    @GetMapping("/{id}")
+    public com.example.userservice.api.RegisteredUser getUser(Integer id) {
+        RegisteredUser user = userService.getUser(id);
+        return new com.example.userservice.api.RegisteredUser(user.getEmail(), user.getName(), user.getSurname());
```

Comment:
For transformation from entity to dto create separate transformer calss (f.e. RegisteredUserTransformer) with static methods that will do transformations in both ways, since in real case scenarios you can have multiple methods that will use this transformer.